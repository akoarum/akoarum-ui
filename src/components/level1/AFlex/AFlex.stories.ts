import { storiesOf } from '@storybook/vue'
import { withKnobs, text, boolean } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import AFlex from './AFlex.vue'

storiesOf('level1/AFlex', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Normal', () => ({
    components: { AFlex },
    props: {
      tag: { default: text('Tag', 'div') },
      direction: { default: text('flex-direction', '') },
      justifyContent: { default: text('justify-content', '') },
      alignItems: { default: text('align-items', '') },
      alignContent: { default: text('align-content', '') },
      wrap: { default: boolean('flex-wrap', true) }
    },
    template: `
      <AFlex
        :tag="tag"
        :direction="direction"
        :justify-content="justifyContent"
        :align-items="alignItems"
        :align-content="alignContent"
        :wrap="wrap"
      >
        <div style="width: 48%; background: #f0f0f0;">テキスト</div>
        <div style="width: 48%; background: #f0f0f0;">テキスト</div>
      </AFlex>
    `
  }), { info: {} })
