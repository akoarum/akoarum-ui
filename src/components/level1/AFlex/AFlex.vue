<script lang="ts">
import { CreateElement, VNode } from 'vue'
import { defineComponent, computed } from '@vue/composition-api'

interface Props {
  tag: string
  direction?: string
  justifyContent?: string
  alignItems?: string
  alignContent?: string
  wrap?: boolean
}

export default defineComponent({
  props: {
    tag: { type: String, default: 'div' },
    direction: { type: String },
    justifyContent: { type: String },
    alignItems: { type: String },
    alignContent: { type: String },
    wrap: { type: Boolean }
  },
  setup(props: Props) {
    const styled = computed<{ [key: string]: string }>(() => {
      const styles: { [key: string]: string } = {}
      if (props.direction) styles['flex-direction'] = props.direction
      if (props.justifyContent) styles['justify-content'] = props.justifyContent
      if (props.alignItems) styles['align-items'] = props.alignItems
      if (props.alignContent) styles['align-content'] = props.alignContent
      if (props.wrap) styles['flex-wrap'] = props.wrap ? 'wrap' : 'nowrap'
      return styles
    })
    return { styled }
  },
  render(createElement: CreateElement): VNode {
    return createElement(
      this.tag,
      {
        class: 'aFlex',
        style: this.styled
      },
      this.$slots.default
    )
  }
})
</script>

<style lang="scss" scoped>
.aFlex {
  display: flex;
}
</style>
