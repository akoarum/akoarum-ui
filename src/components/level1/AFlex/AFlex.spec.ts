import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import AFlex from './AFlex.vue'

describe('AFlex', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(AFlex, {
      propsData: {},
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('styled: スタイルはpropsで制御できる', () => {
      wrapper.setProps({
        justifyContent: 'space-between',
        wrap: true
      })
      expect(wrapper.vm.styled).toEqual({
        'justify-content': 'space-between',
        'flex-wrap': 'wrap'
      })
    })
  })
})
