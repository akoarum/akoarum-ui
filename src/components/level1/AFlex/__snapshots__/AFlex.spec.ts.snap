// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AFlex initialized snapshot 1`] = `
<div
  class="aFlex"
/>
`;
